// angular2-onsenui/dist/src/directives/ons-splitter.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
"use strict";
var import0 = require('./ons-splitter');
var import1 = require('@angular/core/src/change_detection/change_detection_util');
var import3 = require('@angular/core/src/linker/view_utils');
var Wrapper_OnsSplitterSide = (function () {
    function Wrapper_OnsSplitterSide(p0, p1, p2, p3, p4) {
        this._changed = false;
        this.context = new import0.OnsSplitterSide(p0, p1, p2, p3, p4);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_OnsSplitterSide.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_OnsSplitterSide.prototype.ngOnDestroy = function () {
    };
    Wrapper_OnsSplitterSide.prototype.check_page = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.page = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_OnsSplitterSide.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_OnsSplitterSide.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_OnsSplitterSide.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_OnsSplitterSide.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_OnsSplitterSide;
}());
exports.Wrapper_OnsSplitterSide = Wrapper_OnsSplitterSide;
var Wrapper_OnsSplitterContent = (function () {
    function Wrapper_OnsSplitterContent(p0, p1, p2, p3) {
        this._changed = false;
        this.context = new import0.OnsSplitterContent(p0, p1, p2, p3);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_OnsSplitterContent.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_OnsSplitterContent.prototype.ngOnDestroy = function () {
    };
    Wrapper_OnsSplitterContent.prototype.check_page = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.page = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_OnsSplitterContent.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_OnsSplitterContent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_OnsSplitterContent.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_OnsSplitterContent.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_OnsSplitterContent;
}());
exports.Wrapper_OnsSplitterContent = Wrapper_OnsSplitterContent;
