// angular2-onsenui/dist/src/directives/ons-lazy-repeat.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
"use strict";
var import0 = require('./ons-lazy-repeat');
var import1 = require('@angular/core/src/change_detection/change_detection_util');
var import3 = require('@angular/core/src/linker/view_utils');
var Wrapper_OnsLazyRepeat = (function () {
    function Wrapper_OnsLazyRepeat(p0, p1, p2) {
        this._changed = false;
        this.context = new import0.OnsLazyRepeat(p0, p1, p2);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_OnsLazyRepeat.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_OnsLazyRepeat.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
    };
    Wrapper_OnsLazyRepeat.prototype.check_onsLazyRepeatOf = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.onsLazyRepeatOf = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_OnsLazyRepeat.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_OnsLazyRepeat.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_OnsLazyRepeat.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_OnsLazyRepeat.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_OnsLazyRepeat;
}());
exports.Wrapper_OnsLazyRepeat = Wrapper_OnsLazyRepeat;
