// angular2-onsenui/dist/src/directives/ons-pull-hook.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
"use strict";
var import0 = require('./ons-pull-hook');
var Wrapper_OnsPullHook = (function () {
    function Wrapper_OnsPullHook(p0) {
        this._changed = false;
        this.context = new import0.OnsPullHook(p0);
    }
    Wrapper_OnsPullHook.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_OnsPullHook.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
        (this.subscription0 && this.subscription0.unsubscribe());
    };
    Wrapper_OnsPullHook.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_OnsPullHook.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_OnsPullHook.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_OnsPullHook.prototype.subscribe = function (view, _eventHandler, emit0) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.action.subscribe(_eventHandler.bind(view, 'action')));
        }
    };
    return Wrapper_OnsPullHook;
}());
exports.Wrapper_OnsPullHook = Wrapper_OnsPullHook;
