// angular2-onsenui/dist/src/directives/ons-switch.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
"use strict";
var import0 = require('./ons-switch');
var import1 = require('@angular/core/src/change_detection/change_detection_util');
var import3 = require('@angular/core/src/linker/view_utils');
var Wrapper_OnsSwitch = (function () {
    function Wrapper_OnsSwitch(p0) {
        this._changed = false;
        this._changes = {};
        this.context = new import0.OnsSwitch(p0);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_OnsSwitch.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_OnsSwitch.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
        (this.subscription0 && this.subscription0.unsubscribe());
    };
    Wrapper_OnsSwitch.prototype.check_value = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.value = currValue;
            this._changes['value'] = new import1.SimpleChange(this._expr_0, currValue);
            this._expr_0 = currValue;
        }
    };
    Wrapper_OnsSwitch.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        if (!throwOnChange) {
            if (changed) {
                this.context.ngOnChanges(this._changes);
                this._changes = {};
            }
        }
        return changed;
    };
    Wrapper_OnsSwitch.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_OnsSwitch.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_OnsSwitch.prototype.subscribe = function (view, _eventHandler, emit0) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context._valueChange.subscribe(_eventHandler.bind(view, 'valueChange')));
        }
    };
    return Wrapper_OnsSwitch;
}());
exports.Wrapper_OnsSwitch = Wrapper_OnsSwitch;
