// angular2-onsenui/dist/src/directives/ons-navigator.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
"use strict";
var import0 = require('./ons-navigator');
var import1 = require('@angular/core/src/change_detection/change_detection_util');
var import3 = require('@angular/core/src/linker/view_utils');
var Wrapper_OnsNavigator = (function () {
    function Wrapper_OnsNavigator(p0, p1, p2, p3, p4) {
        this._changed = false;
        this.context = new import0.OnsNavigator(p0, p1, p2, p3, p4);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_OnsNavigator.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_OnsNavigator.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
    };
    Wrapper_OnsNavigator.prototype.check_pageComponentType = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.pageComponentType = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_OnsNavigator.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_OnsNavigator.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_OnsNavigator.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_OnsNavigator.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_OnsNavigator;
}());
exports.Wrapper_OnsNavigator = Wrapper_OnsNavigator;
